packages/sign_in_with_apple/sign_in_with_apple/lib/src/sign_in_with_apple.dart

```diff
@@ -1,149 +1,35 @@
 import 'dart:async';
-import 'dart:io';
 
-import 'package:flutter/foundation.dart';
-import 'package:meta/meta.dart';
+import 'package:flutter/material.dart';
 import 'package:flutter/services.dart';
-
-import 'package:sign_in_with_apple/sign_in_with_apple.dart';
-
-// Mention all internal imports implicitly, so we don't forget to export required classes
-import 'package:sign_in_with_apple/src/authorization_credential.dart'
-    show
-        parseAuthorizationCredentialAppleID,
-        parseAuthorizationCredentialAppleIDFromDeeplink,
-        parseAuthorizationCredentialPassword;
-import 'package:sign_in_with_apple/src/credential_state.dart'
-    show parseCredentialState;
+import 'package:sign_in_with_apple_platform_interface/method_channel_sign_in_with_apple.dart';
+import 'package:sign_in_with_apple_platform_interface/sign_in_with_apple_platform_interface.dart';
 
 /// Wrapper class providing the methods to interact with Sign in with Apple.
 // ignore: avoid_classes_with_only_static_members
 class SignInWithApple {
   @visibleForTesting
-  // ignore: public_member_api_docs
-  static const channel = MethodChannel(
-    'com.aboutyou.dart_packages.sign_in_with_apple',
-  );
-
-  /// Returns the credentials stored in the Keychain for the website associated with the current app.
-  ///
-  /// Only available on Apple platforms.
-  ///
-  /// Throws a [SignInWithAppleException] exception when no credentials have been found in the Keychain.
-  static Future<AuthorizationCredentialPassword> getKeychainCredential() async {
-    try {
-      if (!Platform.isIOS &&
-          !Platform.isMacOS &&
-          Platform.environment['FLUTTER_TEST'] != 'true') {
-        throw const SignInWithAppleNotSupportedException(
-          message: 'The current platform is not supported',
-        );
-      }
+  static MethodChannel get channel =>
+      (SignInWithApplePlatform.instance as MethodChannelSignInWithApple)
+          // ignore: invalid_use_of_visible_for_testing_member
+          .channel;
 
-      final response = await channel.invokeMethod<Map<dynamic, dynamic>>(
-        'performAuthorizationRequest',
-        [
-          const PasswordAuthorizationRequest(),
-        ].map((request) => request.toJson()).toList(),
-      );
-
-      if (response == null) {
-        throw Exception('getKeychainCredential: Received `null` response');
-      }
-
-      return parseAuthorizationCredentialPassword(response);
-    } on PlatformException catch (exception) {
-      throw SignInWithAppleException.fromPlatformException(exception);
-    }
+  static Future<bool> isAvailable() async {
```

Comment:
Right now we would lose the documentation on these methods (in the Editor and on pub.dev). Should we maybe make the methods non-static and then use `implements SignInWithAppleInterface`?

Reply:
I shuffled all the docs around in #263 so we get them inline in the editor again.